src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -107,6 +117,10 @@ public SearchResultPageDTO<Lecture> getAllOnPageWithSize(final PageableSearchDTO
      * @param lecture the lecture to be deleted
      */
     public void delete(Lecture lecture) {
+        Channel lectureChannel = channelRepository.findChannelByLectureId(lecture.getId());
+        if (lectureChannel != null) {
+            conversationService.deleteConversation(lectureChannel);
```

Comment:
This is synchronous call and it deletes all participants and all posts in that conversation, do you think its a good idea to make this call `async`? Same applies for others.

The cost of the delete operation in deleteConversation can increase with the number of posts and participants in a conversation. If there are many posts and participants, the overall delete operation can be more expensive.

Reply:
Good point. I made it async

Reply:
I talked to Stephan and we decided to delete synchronously.